unison-core/src/Unison/Name.hs

```diff
@@ -160,18 +161,32 @@ endsWithReverseSegments :: Name -> [NameSegment] -> Bool
 endsWithReverseSegments (Name _ ss0) ss1 =
   List.NonEmpty.isPrefixOf ss1 ss0
 
--- >>> stripReversedPrefix "a.b.c" ["b", "a"]
--- Just c
--- >>> stripReversedPrefix "x.y" ["b", "a"]
--- Nothing
--- >>> stripReversedPrefix "a.b" ["b", "a"]
+-- >>> stripReversedPrefix (fromReverseSegments ("c" :| ["b", "a"])) ["b", "a"]
+-- Just (Name Relative (NameSegment {toText = "c"} :| []))
+-- >>> stripReversedPrefix (fromReverseSegments ("y" :| ["x"])) ["b", "a"]
 -- Nothing
+--
+-- >>> stripReversedPrefix (fromReverseSegments ("c" :| ["b", "a"])) ["b", "a"]
+-- Just (Name Relative (NameSegment {toText = "c"} :| []))
```

Comment:
These updated doctests seem harder to read? What's the idea here?

Reply:
The FromString and nice 'show' instance were removed when Mitchell moved them into the Syntax module and the doctests just hadn't been re-generated since then.

Reply:
I kinda still want to fix this because I think the new tests are too hard to read. @mitchellwrosen wdyt about the imports/organization issue?

Reply:
I would say if we want nicer syntax we could move these tests to a test module with orphan FromString and Show instances